### PR TITLE
FIX-#2093: Add ability displaying of method's name for inplace operations by progress bar

### LINKED ISSUE #2093: 'update' function implementation
### Describe the problem

While working with @devin-petersohn on implementing a progress bar, it seemed like the implementation of df.update() or other in-place operations might want to use different functions or names. Currently the progress bar says computation is being performed by a "caller" function.

### Source code / logs

image

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2093  
- [x] tests passing

### CODE REVIEW

**Review by devin-petersohn — COMMENTED** [review]
Let's push this one to the next release since the progress bar won't make it either

File: modin/backends/pandas/query_compiler.py [hunk 1]
```diff
@@ -123,14 +123,16 @@ def dt_op_builder(df, *args, **kwargs):
     return dt_op_builder
 
 
-def copy_df_for_func(func):
+def copy_df_for_func(func, func_name=None):
```
Comment (vnlitvinov) [inline modin/backends/pandas/query_compiler.py:126]:
```suggestion
def copy_df_for_func(func, func_name:str=None):
```
with whatever formatting `black` wants

File: modin/backends/pandas/query_compiler.py [hunk 2]
```diff
@@ -123,14 +123,16 @@ def dt_op_builder(df, *args, **kwargs):
...
     """
     Create a function that copies the dataframe, likely because `func` is inplace.
 
     Parameters
-    ----------
+    ---------
```
Comment (vnlitvinov) [inline modin/backends/pandas/query_compiler.py:131]: why?

File: modin/backends/pandas/query_compiler.py (hunk 1 above)
Comment (YarShev) [reply modin/backends/pandas/query_compiler.py:126]: Okay, will be fixed.

File: modin/backends/pandas/query_compiler.py (hunk 2 above)
Comment (YarShev) [reply modin/backends/pandas/query_compiler.py:131]: It is unintentionally. Thanks for notice!

**Review by vnlitvinov — DISMISSED** [review]

**Comment by vnlitvinov** [comment]
@devin-petersohn 0.8.1 was cut, I guess we should merge this?

File: modin/backends/pandas/query_compiler.py [hunk 3]
```diff
@@ -123,14 +123,16 @@ def dt_op_builder(df, *args, **kwargs):
     return dt_op_builder
 
 
-def copy_df_for_func(func):
+def copy_df_for_func(func, func_name: str = None):
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:126]:
```suggestion
def copy_df_for_func(func, display_name: str = None):
```

File: modin/backends/pandas/query_compiler.py [hunk 4]
```diff
@@ -123,14 +123,16 @@ def dt_op_builder(df, *args, **kwargs):
...
     """
     Create a function that copies the dataframe, likely because `func` is inplace.
 
     Parameters
     ----------
     func : callable
         The function, usually updates a dataframe inplace.
+    func_name : str, optional
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:134]:
```suggestion
    display_name : str, optional
```

File: modin/backends/pandas/query_compiler.py [hunk 5]
```diff
@@ -143,6 +145,8 @@ def caller(df, *args, **kwargs):
         func(df, *args, **kwargs)
         return df
 
+    if func_name is not None:
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:148]:
```suggestion
    if display_name is not None:
```

File: modin/backends/pandas/query_compiler.py [hunk 6]
```diff
@@ -332,11 +336,12 @@ def to_numpy(self, **kwargs):
     __rxor__ = BinaryFunction.register(pandas.DataFrame.__rxor__)
     __xor__ = BinaryFunction.register(pandas.DataFrame.__xor__)
     df_update = BinaryFunction.register(
-        copy_df_for_func(pandas.DataFrame.update), join_type="left"
+        copy_df_for_func(pandas.DataFrame.update, func_name="update"), join_type="left"
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:339]:
```suggestion
        copy_df_for_func(pandas.DataFrame.update, display_name="update"), join_type="left"
```

File: modin/backends/pandas/query_compiler.py [hunk 7]
```diff
@@ -332,11 +336,12 @@ def to_numpy(self, **kwargs):
...
     )
     series_update = BinaryFunction.register(
         copy_df_for_func(
-            lambda x, y: pandas.Series.update(x.squeeze(axis=1), y.squeeze(axis=1))
+            lambda x, y: pandas.Series.update(x.squeeze(axis=1), y.squeeze(axis=1)),
+            func_name="update",
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:344]:
```suggestion
            display_name="update",
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
Minor renaming change request for clarity.

**Comment by YarShev** [comment]
@devin-petersohn , suggestions have been applied.

**Comment by YarShev** [comment]
@devin-petersohn , just a friendly reminder to review.

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @YarShev

**Pull request merged**